src/publisher/fragment_logic.py

```diff
@@ -326,10 +326,14 @@ def set_article_json(av, data=None, quiet=True, hash_check=True, update_fragment
     oldhash = av.article_json_hash
     newhash = hash_ajson(result)
 
-    if (
-        hash_check
-        and result
-        and _identical_articles(raw_original, raw_new, result, oldhash, newhash)
+    # todo: more bad old logic that silently fails on validation errors. remove.
+    if not result or not snippet:
```

Comment:
rm comment, it was for a different bit of code.

check for tests/coverage around this, especially when quiet=True (forced ingest). I suspect it's redundant.

Reply:
yeah, no tests are touching it. So ... a silent correction (forced ingest) that creates invalid article-json should trigger this.

Reply:
edit - I was mistaken, coverage wasn't setup properly for ad-hoc testing. all is well.